### PR TITLE
[FIX] website_sale_mondialrelay: adapt to new checkout

### PR DESCRIPTION
**Pull request by vchu-odoo**
It was possible to use delivery as billing address even if delivery address was a modialrelay one. That caused a traceback as it is not allowed to change modialrelay address. This commit will prevent these cases.

### CODE REVIEW

File: addons/website_sale_mondialrelay/static/src/js/checkout.js [hunk 1]
```diff
@@ -23,26 +23,80 @@ WebsiteSaleCheckout.include({
...
-        this._super(...arguments);
-        if (result.mondial_relay) {
+    async _selectDeliveryMethod(ev) {
+        const checkedRadio = ev.currentTarget;
+        await this._super(...arguments);
+        this.use_delivery_as_billing_toggle.disabled = this._hasMondialrelaySelected();
+        if (checkedRadio.dataset.isMondialrelay) {
+            // Uncheck use same as delivery and show the billing address row.
+            this.use_delivery_as_billing_toggle.checked = false;
+            this.billingContainer.classList.remove('d-none');
```
Comment (AntoineVDV) [inline addons/website_sale_mondialrelay/static/src/js/checkout.js:50]: Why not call the toggle method?

File: addons/website_sale_mondialrelay/static/src/js/checkout.js [hunk 2]
```diff
@@ -23,26 +23,80 @@ WebsiteSaleCheckout.include({
...
+     * @override of `website_sale`
+     */
+    async _changeAddress(ev) {
+        const newAddress = ev.currentTarget;
+        if (newAddress.dataset.isMondialrelay && this.use_delivery_as_billing_toggle.checked) {
+            // Uncheck use same as delivery and show the billing address row.
+            this.use_delivery_as_billing_toggle.checked = false;
+            this.billingContainer.classList.remove('d-none');
+        }
+        this._super(...arguments);
```
Comment (AntoineVDV) [inline addons/website_sale_mondialrelay/static/src/js/checkout.js:78]:
```suggestion
        await this._super(...arguments);
```

File: addons/website_sale_mondialrelay/static/src/js/checkout.js [hunk 3]
```diff
@@ -23,26 +23,80 @@ WebsiteSaleCheckout.include({
...
+     * row. Mondialrelay addresses are not allowed to be selected as billing.
+     *
+     * @override of `website_sale`
+     */
+    async _changeAddress(ev) {
+        const newAddress = ev.currentTarget;
+        if (newAddress.dataset.isMondialrelay && this.use_delivery_as_billing_toggle.checked) {
+            // Uncheck use same as delivery and show the billing address row.
+            this.use_delivery_as_billing_toggle.checked = false;
+            this.billingContainer.classList.remove('d-none');
```
Comment (AntoineVDV) [inline addons/website_sale_mondialrelay/static/src/js/checkout.js:76]: Call the toggle method here also?

**Review by AntoineVDV — CHANGES_REQUESTED** [review]

**Review by AntoineVDV — APPROVED** [review]

**Comment by d-fence** [comment]
Upgrade exception [#385](https://runbot.odoo.com/web/#id=385&view_type=form&model=runbot.upgrade.exception) added.

Waiting the forward-port of this pr, the exception will be applied on all builds, and create an inconsistent state for migrations.
**Please forward port this asap up to master without change.** Exception should be forward-ported before the end of the week.

PRs adding fields/models should be merged at the beginning of the week and all the forward ports should be merged in the same week.

If you need to apply any change before it reaches master, please notify runbot team.

Details:
```
view:website_sale_mondialrelay.website_sale_mondialrelay_billing_address_row
view:website_sale_mondialrelay.mondialrelay_delivery_method
```
cc @KangOl @nseinlet @aj-fuentes

**Comment by AntoineVDV** [comment]
@robodoo r+ delegate+

**Pull request closed without merging**